### ISSUE TITLE
Don't allow organization to be empty for documents

### ISSUE THREAD

**Issue by pcambra** [humanitarianresponse/site#65]
When creating a new document, organization is a mandatory field but if you fill it with something without waiting for the autocomplete, or if you input free text that is not an organization, the system will submit and save an empty organization.

See https://www.drupal.org/node/1819618

Javed's comments:

"someone uploaded a document https://www.humanitarianresponse.info/en/operations/burkina-faso/document/nutrition-causal-analysis-link-nca-tapoa-burkina-faso-2013 but the organization part was missing which I just tagged (ACF). Org field is mandatory but still the content is saved in the website."

"using the API(https://www.humanitarianresponse.info/api/v1.0/documents?fields=label,organizations), i see quite a lot of documents with 'null' organizations. I guess we will have to do some clean up which will also require some back and forth with the space owners to have the org assigned to those documents."

We need to check if this has happened in other content types too.

**Comment by javedyousafzai** [comment]
Yes if I check the maps via API (https://www.humanitarianresponse.info/api/v1.0/infographics?fields=label,organizations), I see organizations with 'null' values.

**Comment by javedyousafzai** [comment]
just trying to understand this issue better, If someone uploaded a document with org 'ABC-test' and it didn't exist in the HR.info, the organization went to null right?

**Comment by pcambra** [comment]
That's exactly right @javedyousafzai

**Comment by javedyousafzai** [comment]
ouch :) so now we won't know which organization they assigned to the documents.

**Comment by pcambra** [comment]
This issue has 2 parts. The first one is fix the validation and ensure that this doesn't happen anymore.
The second part is identify all the documents (potentially other node types can be affected) with no organization and remove those coming from hr 1.x.
Not sure how many these might be but we can study what we do with these once we have a listing.

**Comment by pcambra** [comment]
Remaining tasks:

- Create a spreadsheet with a list of content with empty organisation with their author name, email and created date. (Pedro)
- Review the list and contact the relevant organisations. (Javed)
- Consider manual or bulk updates for fixing this.

**Comment by pcambra** [comment]
Created a gist to generate this info if we need it in the future https://gist.github.com/pcambra/72b575d53d42ae22b0ce

Sending an email with the file

**Comment by javedyousafzai** [comment]
If I ask the site focal points to enter organization names in the excel sheet, would you be able to import them and update the records in the site? This will be the most easy option as there are lots of content whose title can tell which organizations they belong to.

I am just trying to finds a fast way as there are lots of content without organizaitons.

**Comment by javedyousafzai** [comment]
OK here it goes. 

In order to tag organization against the content, the focal point has to copy the url from excel, and then edit the content in the site which will take quite some time.

Instead of doing above, what if the person enter the organization name directly in the excel sheet. Lets say 50% content get the organization this way which will be very fast - can we then import this excel sheet in HR.info so the content/record is updated in the website??

**Comment by pcambra** [comment]
Yes, we can, just create a new column in the CSV file and we can process it.

**Comment by javedyousafzai** [comment]
sweet.

thanks

**Comment by javedyousafzai** [comment]
@pcambra as discuss in our call today. it would be great if you can add operation name, cluster site and update date columns in the csv output.

**Comment by pcambra** [comment]
Sent, was working on it
